test/s3path.jl

```diff
@@ -354,4 +355,34 @@ end
     rm(json_path)
 end
 
+@testset "S3Path versioning" begin
+    key_version_file = "test_versions"
+    s3_put(bucket_name, key_version_file, "data.v1")
+    s3_put(bucket_name, key_version_file, "data.v2")
+
+    # `s3_list_versions` returns versions in the order newest to oldest
+    versions = reverse(d["VersionId"] for d in s3_list_versions(aws, bucket_name, key_version_file))
```

Comment:
My bad: `reverse` doesn't work on generators
```suggestion
    versions = [d["VersionId"] for d in reverse!(s3_list_versions(aws, bucket_name, key_version_file))]
```